src/App.tsx

```diff
@@ -13,30 +15,11 @@ const styles = StyleSheet.create({
 });
 
 const App = () => {
-  const [currentTime] = React.useState<Date>(new Date());
-
-  function isItCoffeeTime() {
-    if (
-      (currentTime.getHours() >= 8 && currentTime.getHours() <= 9) ||
-      (currentTime.getHours() >= 12 && currentTime.getHours() <= 13) ||
-      currentTime.getTime() >= currentTime.setHours(17, 30)
-    ) {
-      return false;
-    }
-    return true;
-  }
-
   return (
     <SafeAreaView style={styles.safeArea}>
       <ScrollView contentContainerStyle={styles.scrollView}>
-        <Text>
-          Time:{' '}
-          {currentTime.toLocaleString('en-us', {
-            hour: 'numeric',
-            minute: '2-digit',
-          })}
-        </Text>
-        <Text>Is it coffee time? {isItCoffeeTime() ? 'Yes' : 'No'}</Text>
+        <CurrentTime />
+        <IsItCoffeeTime />
```

Comment:
Good intuition!

Generally, components are reusable building blocks or view containers.

It's also worth noting that often times we need to "connect" these components to data. This data may come from network fetches to an API, from cache, from local datastores, state containers, etc. When starting to build out tests, it becomes increasingly difficult to write them against these connected components... so two things are worth taking into consideration:
1) Separating "Connected" components from "Presentational" components to ease testing. A connected component deals with fetching the data, processing, error handling etc...and a presentational component is purely responsible for rendering the data and/or UI elements.
2) Fixtures and mocks